controllers/swiftring_controller.go

```diff
@@ -160,6 +160,10 @@ func (r *SwiftRingReconciler) reconcileNormal(ctx context.Context, instance *swi
 
 	deviceList, deviceListHash, err := swiftring.DeviceList(ctx, helper, instance)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info(fmt.Sprintf("%s... requeueing", err.Error()))
+			return ctrl.Result{RequeueAfter: time.Duration(5) * time.Second}, nil
```

Comment:
Couldn't this introduce a problem?
Because you are no longer setting the condition to False in this instance, so it will never change from True to False even if the conditions say it should.

Shouldn't this be added right after setting the condition?

Reply:
The status update below is setting an error state - but I think this is not a "real" error, it just means expected PVCs are not found. For "real" errors the status is still updated.
Do you think there should be a state inbetween? Thinking more about this, maybe adding an InputReadyCondition as a follow up might be a good idea, and then setting this to InputReadyWaitingMessage

Reply:
I thought that conditions were meant to be set to False until they are met, and in your case until you have the swiftring device list you have to set the SwiftRingReadyCondition to False, maybe you just need to have 2 different messages in the condition, one where you set it to False and say you are waiting for it to appear and another where you say False and the error you see.

But not being able to transition from True to False if a human operator deletes the swiftring object from etcd seems like a bug to me.

Reply:
I get your point - I did a couple more test and watched the Status changes, and this is an issue.

I'll move the "if apierrors.IsNotFound(err)..." after the status update, and once openstack-k8s-operators/lib-common/#477 is merged will add a new PR to use InputReadyCondition and JobReadyCondition to report more details in the status (without actually reporting an error when there is none).

Reply:
For the record, this is the status progress as of now:

Initial deployment:
```
swift-ring   Unknown   Setup started
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-0" not found
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-0" not found
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-0" not found
swift-ring   True      Setup complete
```

Upscaling:
```
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-1" not found
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-1" not found
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-1" not found
swift-ring   True      Setup complete
```

I think using "Input data resources missing" and adding "Job in progress" makes sense, thus /openstack-k8s-operators/lib-common#477 will be useful in a followup.